core/state_processor.go

```diff
@@ -407,8 +407,7 @@ func (p *StateProcessor) Process(block *types.WorkObject) (types.Receipts, []*ty
 					if primeTerminus == nil {
 						return nil, nil, nil, nil, 0, fmt.Errorf("could not find prime terminus header %032x", header.PrimeTerminus())
 					}
-					// Convert Qi to Quai
-					msg.SetValue(misc.QiToQuai(primeTerminus, etx.Value()))
+					msg.SetValue(misc.QiToQuai(primeTerminus, types.Denominations[uint8(tx.Value().Uint64())], p.engine.DiffToBigBits(primeTerminus)))
```

Comment:
see here you are using the prime terminus

Reply:
Also how does `types.Denominations[uint(tx.Value().Uint64())]` translate to the qi amount? does that mean you can only set a qiAmt as a denomination value ? because value can be anything right? we have to split the denominations and make sense of it?